### PR TITLE
Import most recent MS3 additions

### PR DESCRIPTION
**Pull request by jam01**
Addresses #59

### CODE REVIEW

File: docs/modules/ROOT/pages/libraries-crypto.adoc [hunk 1]
```diff
@@ -1,5 +1,35 @@
 ## crypto
 
+### `decrypt(string value, string secret, string algorithm, string mode, string padding)`
```
Comment (JakeMHughes) [inline docs/modules/ROOT/pages/libraries-crypto.adoc:3]: docs need to be updated with your changes to the functions

Comment (jam01) [reply docs/modules/ROOT/pages/libraries-crypto.adoc:3]: ah yeah

File: src/test/java/com/datasonnet/NestedDocumentsTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        Map<String, Document<String>> nested = new HashMap<>(2);
+        nested.put("xml", new DefaultDocument<>(xml, MediaTypes.APPLICATION_XML));
+        nested.put("json", new DefaultDocument<>(json, MediaTypes.APPLICATION_JSON));
+
+        Map<String, Document<?>> inputs = Collections.singletonMap("nested", new DefaultDocument<>(nested, MediaTypes.APPLICATION_JAVA));
+        String result = new MapperBuilder("nested")
+                .withInputNames("nested")
+                .build()
+                .transform(DefaultDocument.NULL_INSTANCE, inputs, MediaTypes.APPLICATION_JSON)
```
Comment (fugu13) [inline src/test/java/com/datasonnet/NestedDocumentsTest.java:44]: I like this test, but it feels like it is intended to test the multiple inputs as a map in the input functionality, and it doesn't currently--instead it uses the preexisting secondary inputs capability (the map of inputs in the second argument).

Let's keep this test and add another test in this set of tests that runs a similar transformation but with a map passed as the main input, to exercise that new functionality

**Review by fugu13 — CHANGES_REQUESTED** [review]
Jake's already covered the docs issue, the only thing I see is it could use one more test covering the multiple inputs in a map functionality

File: src/test/java/com/datasonnet/NestedDocumentsTest.java (hunk 2 above)
Comment (jam01) [reply src/test/java/com/datasonnet/NestedDocumentsTest.java:44]: it's really only meant for inputs, and not for the payload

**Pull request merged**